ethcore/src/client.rs

```diff
@@ -455,6 +472,59 @@ impl BlockChainClient for Client {
 			best_block_number: From::from(chain.best_block_number())
 		}
 	}
+
+	fn blocks_with_bloom(&self, bloom: &H2048, from_block: BlockId, to_block: BlockId) -> Option<Vec<BlockNumber>> {
+		match (self.block_number(from_block), self.block_number(to_block)) {
+			(Some(from), Some(to)) => Some(self.chain.read().unwrap().blocks_with_bloom(bloom, from, to)),
+			_ => None
+		}
+	}
+
+	fn logs(&self, filter: Filter) -> Vec<LocalizedLogEntry> {
+		let mut blocks = filter.bloom_possibilities().iter()
+			.map(|bloom| self.blocks_with_bloom(bloom, filter.from_block.clone(), filter.to_block.clone()))
+			.filter_map(|m| m)
+			.flat_map(|m| m)
+			// remove duplicate elements
+			.collect::<HashSet<u64>>()
```

Comment:
`itertools` is not ethcore direct dependency and I didn't want to introduce it just because of this one place. Besides that, itertools `unique` implementation does exactly the same thing as this code - it puts everything in `HashSet`